lazy_tensor_core/test/cpp/test_aten_ltc_ts_tensor.cpp

```diff
@@ -2858,6 +2858,17 @@ TEST_F(AtenLtcTsTensorTest, TestEmpty) {
   });
 }
 
+TEST_F(AtenLtcTsTensorTest, TestZeroInPlace) {
+  torch::Tensor a = torch::zeros({2, 2}, torch::TensorOptions(torch::kFloat));
+
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor b = torch::ones({2, 2}, torch::TensorOptions(torch::kFloat));
+    torch::Tensor xla_b = CopyToDevice(b, device);
+    xla_b.zero_();
```

Comment:
One thing I have been confused for a while, is there any checks I can do to make sure that the operator actually goes through the TS route here?

Reply:
What I did was gdb into one test to understand the logic. In terms of checking, I guess you can add device checking for those `xla_*` tensors.

Reply:
Thanks for the device checking pro-tip. May I ask exactly how? Do we have a helper here already?

Reply:
I think it's moderately beneficial to automate device checking on the returned tensor (mainly to make sure we didn't _totally_ drop the ball and return a non-lazy tensor from a lazy-tensor method)

But in practice, there could have been an eager fallback rather than a TS lowering and the tensor would still come back on the lazy device because of how fallback works.  

I think the only way to really check is to rely on instrumentation that is not part of the public API- in practice, it's fairly easy to just dump the TS IR graph or GDB  or println, but we could see if we can add a better instrumentation API for making tests (or debuginfo) easier to write